src/ctia/entity/feed.clj

```diff
@@ -0,0 +1,239 @@
+
+(s/defschema FeedSearchParams
+  (st/merge
+   {:query s/Str}
+   PagingParams
+   BaseEntityFilterParams
+   FeedFieldsParam
+   {(s/optional-key :sort_by) feed-sort-fields}))
+
+(def FeedGetParams FeedFieldsParam)
+
+(s/defschema FeedListQueryParams
+  (st/merge
+   PagingParams
+   FeedFieldsParam
+   {(s/optional-key :sort_by) feed-sort-fields}))
+
+(s/defschema FeedByExternalIdQueryParams
+  FeedListQueryParams)
+
+(s/defschema FeedView
+  {(s/optional-key :judgements) [Judgement]
+   (s/optional-key :observables) [Observable]})
+
+(def identity-map
+  {:authorized-anonymous true})
+
+(defn valid-lifetime? [lifetime]
+  (cdv/valid-now?
+   (java.util.Date.)
+   {:valid_time lifetime}))
+
+(defn fetch-feed [id s]
+  (if-let [{:keys [indicator_id
+                   secret
+                   output
+                   lifetime
+                   owner
+                   groups]}
+           (read-store :feed
+                       read-record
+                       id
+                       identity-map
+                       {})]
+
+    (cond
+      (not (valid-lifetime? lifetime)) :not-found
+      (not= s secret) :unauthorized
+      :else (let [;; VERY IMPORTANT! inherit the identity from the Feed!
+                  feed-identity
+                  {:login owner
+                   :groups groups
+                   :capabilities #{:read-judgement
+                                   :read-relationship
+                                   :list-relationships}}
+                  feed-results
+                  (some->> (list-all-pages
+                            :relationship
+                            list-records
+                            {:all-of {:target_ref indicator_id}}
+                            feed-identity
+                            {:fields [:source_ref]})
+                           (map :source_ref)
+                           (remove nil?)
+                           (map #(read-store :judgement
+                                             read-record
+                                             %
+                                             feed-identity
+                                             {}))
+                           (remove nil?)
+                           (map with-long-id))]
+
+              (cond-> {}
+                (= :observables output)
+                (assoc
+                 :output :observables
+                 :observables
+                 (distinct (map :observable
+                                feed-results)))
+                (= :judgements output)
+                (assoc
+                 :output :judgements
+                 :judgements
+                 (distinct (map un-store
+                                feed-results))))))
+    :not-found))
+
+(defn sorted-observable-values [data]
+  (sort-by :value
+           (map #(select-keys % [:value]) data)))
+
+(defn render-headers? [output]
+  (not= :observables output))
```

Comment:
I don't like the fact that headers are rendered or not based on the output type. Could you add a query param `render-headers`?

Reply:
I think we shouldn't add query parameters for this route because we may switch to generated views if we get performance issues.